lib/puma/server.rb

```diff
@@ -28,6 +28,9 @@ class Server
     include Puma::Const
     extend  Puma::Delegation
 
+    # TCP_TIME_WAIT: 6, TCP_CLOSE: 7, TCP_CLOSE_WAIT: 8, TCP_LAST_ACK: 9
+    IGNORED_CHECK_CLIENT_SOCKET_STATES = (6..9)
```

Comment:
I haven't done prolonged testing for this, however, if, say, your TCP backlog is 1000 (or whatever) it seems possible that something at the end of the backlog could go through a full closing of the socket while it's still in the backlog. To be honest I'm not entirely sure of the mechanics here in Linux' TCP/IP implementation of whether you "have" to `accept(2)` and `close(2)` it from the listening socket (kind of like `wait(2)` on a zombie process), or if you could ever accept a socket that's already in this state. I mostly included it for completeness of socket states we definitely _don't_ care about. I found that we're also missing the `TCP_CLOSING` (server has buffered data that it hasn't sent yet but the other end has closed). I don't see how this would ever happen in Puma, but it's also not a socket state we're interested in. For the Unicorn patch we'll be running in production in the coming weeks, I may actually patch it so that we print the socket state that we rejected. It'd be interesting to see over a month if it's _always_ `CLOSE_WAIT` (which it has been in my testing), or if it is ever one of the others. I haven't nerded out with `tcpdump` and `netstat` to be completely confident it can ever end up in the other states. If you think it's key, we can certainly go with only `CLOSE_WAIT`—since it's the only one I've observed in testing.